.github/workflows/workspace_integration_test.yaml

```diff
@@ -1,12 +1,79 @@
+    secrets:
+      studio_license_key:
+        required: false
 
 jobs:
-  job_name:
-    name: Any job name with
+  setup:
     runs-on: ubuntu-latest
+    outputs:
+      #We return the image_tag input of the workflow_call or the dotenv action if it is not provided
+      image_tag: ${{ github.event.inputs.image_tag || steps.dotenv.outputs.studio_docker_tag }}
     steps:
-      - name: Hello World
-        run: echo "Hello World after"
\ No newline at end of file
+      - uses: actions/checkout@v4
+      - id: dotenv
+        uses: falti/dotenv-action@v1.0.4
+  integration-test-in-studio-container:
+    needs: setup
+    runs-on: ubuntu-latest
+    env:
+      STUDIO_LICENSE_KEY: ${{ secrets.studio_license_key || secrets.STUDIO_CI_LICENSE_KEY }}
+      # TODO this variable is required but github workspace is different
+      # see https://github.com/actions/runner/issues/2058
+      USER_WS: /github/workspace
+      ROS_DOMAIN_ID: 49
+      ROS_LOCALHOST_ONLY: 1
+      RMW_IMPLEMENTATION: rmw_cyclonedds_cpp
+      CYCLONEDDS_URI: /tmp/cyclonedds.xml
+    # TODO: docker tag
+    # TODO: example for custom base images
+    container: picknikciuser/moveit-studio:${{ needs.setup.outputs.image_tag }}
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          submodules: recursive
+      # The 'lo' interfaces will not have multicast, and will fail trying to claim a participant index
+      # with the default cyclone dds config. We set one for the unit tests to be able to execute.
+      - name: Configure DDS
+        run: |
+          cat <<EOF > /tmp/cyclonedds.xml
+          <?xml version="1.0" ?>
+          <CycloneDDS xmlns="https://cdds.io/config" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="https://cdds.io/config https://raw.githubusercontent.com/eclipse-cyclonedds/cyclonedds/master/etc/cyclonedds.xsd">
+            <Domain id="any">
+              <Discovery>
+                <ParticipantIndex>auto</ParticipantIndex>
+                <MaxAutoParticipantIndex>120</MaxAutoParticipantIndex>
+              </Discovery>
+            </Domain>
+          </CycloneDDS>
+          EOF
+      - name: Install rosdeps
+        run: |
+          . /opt/ros/humble/setup.sh
+          . /opt/underlay_ws/install/setup.sh
+          . /opt/overlay_ws/install/setup.sh
+          apt-get update
+          rosdep update
+          rosdep install --from-paths src --ignore-src -r -y
+      - name: Run colcon build
+        run: |
+          . /opt/ros/humble/setup.sh
+          . /opt/underlay_ws/install/setup.sh
+          . /opt/overlay_ws/install/setup.sh
+          colcon build
+      - name: Run colcon test
+        run: |
+          . /opt/ros/humble/setup.sh
+          . /opt/underlay_ws/install/setup.sh
+          . /opt/overlay_ws/install/setup.sh
+          . ./install/setup.sh
+          colcon test --retest-until-pass 3
+      - if: always()
+        run: |
+          colcon test-result --verbose
```

Comment:
nit: add newline at the end

Reply:
https://github.com/PickNikRobotics/reusable_ci/commit/572d25f9e8c5bfa225ed5f644f87bbf737d32dbf